source/accessories/amds/firmware/index.md

```diff
@@ -105,8 +102,8 @@ The AMDS firmware works, albeit with limitations as described above. Some ideas
 
 1. The AMDS cannot be configured from the master. Improvements could use an additional TX/RX pair to enable simple register protocol for config. This could be used to set digital filter bandwidths, turn on/off sensor card slots for faster sampling, etc.
 
-2. The `SYNC_TX` and `SYNC_ADC` ISR priorities should probably be flipped. In other words, the ADC sampling should be highest priority. This would mean that the sampling would never miss alignment with the PWM carrier. However, this places a burden on the master since it must turn off requests for ADC sampling when it wants to receive new data. If the master kept telling the AMDS to sample the ADCs at all times, it could never send the data back since the sampling ISR would be highest priority.
+2. The ADC sampling throughput could be improved above 280kHz. Theoretically, the ADC devices support upwards of 1Msps, or 500ksps in two device daisy chain. This would probably require shortening the delay when the ADC is doing the sampling. The latest AMDS mainboard provides `BUSY` signals for each ADC which can be used for ISRs to end the ADC sampling window. These are not used in the current firmware. Instead, the simpler approach of busy waiting until the max timeout occurs is used (i.e. wait for 1300ns). However, the nominal wait time is only about 50% of this.
```

Comment:
@npetersen2 
This is just something you had previously written; it's on the currently published docs. It got marked as an addition because deleted the previous `2.` which was about the SYNC_TX signal and made this the new `2.`  I'm not sure I understand what you mean, because this is about the individual ADC sampling rate. I can change it if you want. What should I say instead?

Reply:
Ha, classic, I complain about my own previous work ;)

I think we should revise this a little bit. In the previous firmware, the ADC sampling and the TX transmission were different, so it made sense to discuss separately... but now, they are one thing.

Let's change it to say:

> 2. The total sampling and data transmission latency as seen by the master (AMDC) could be improved. In the current firmware, the time from when the `SYNC_ADC` line is toggled to when the newly sampled data has fully been received by the master is about 11 µs. This places limitations on the control rate of the master (AMDC). The improvements can come in several parts: 1) the actual ADC sampling time can be shorten by using the `BUSY` signal from each ADC in an ISR to end the ADC sampling window. These are not used in the current firmware. Instead, the simpler approach of busy waiting until the max timeout occurs is used (i.e. wait for 1300ns). However, the nominal wait time is only about 50% of this. 2) The data transmission time could be shorten by removing/reducing the overhead in the packet format, i.e., removing the header bytes.